lib/helpers.js

```diff
@@ -127,13 +132,56 @@ const parseResults = (editor, results, filePath, showIgnored) => {
   return toReturn;
 };
 
+function requireCached(request) {
+  const modulePath = require.resolve(request);
+  if (moduleCache.has(modulePath)) {
+    return moduleCache.get(modulePath);
+  }
+  // eslint-disable-next-line import/no-dynamic-require
+  const moduleInstance = require(modulePath);
+  moduleCache.set(modulePath, moduleInstance);
+  return moduleInstance;
+}
+
+export function getStylelintPath(filePath) {
+  try {
+    return resolve('stylelint', { basedir: path.dirname(filePath) });
+  } catch (e) {
+    return stylelintLocalPath;
+  }
+}
+
+export function refreshModulesPath(modulesDir) {
+  if (lastModulesPath !== modulesDir) {
+    lastModulesPath = modulesDir;
+    process.env.NODE_PATH = modulesDir || '';
+    // eslint-disable-next-line no-underscore-dangle
+    require('module').Module._initPaths();
+  }
+}
+
+export function getStylelintInstance(filePath) {
+  const stylelintPath = getStylelintPath(filePath);
+  // stylelintPath ends with 'node_modules/stylelint/lib/index.js' and we need
+  // just the 'node_modules' path
+  const modulesDir = path.dirname(path.dirname(path.dirname(stylelintPath)));
+  refreshModulesPath(modulesDir);
+  return requireCached(stylelintPath);
+}
+
 export const runStylelint = async (editor, stylelintOptions, filePath, settings) => {
   startMeasure('linter-stylelint: Stylelint');
   let data;
+  // store current NODE_PATH so it can be restored when linting completes.
+  originalNodePath = process.env.NODE_PATH;
+  const instance = getStylelintInstance(filePath);
```

Comment:
The `stylelint` instance should be based on the project within Atom, not the specific file, correct? Elsewhere we use this code to determine the project root path, falling back to the directory of the file if there is no project open:
```js
let projectDir = atom.project.relativizePath(filePath)[0];
if (projectDir == null) {
  projectDir = path.dirname(filePath);
}
```